src/main/java/book/store/onlinebookstore/controller/BookController.java

```diff
@@ -21,37 +24,48 @@
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("/api/books")
+@Tag(name = "Book management", description = "Endpoints for managing books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get book by id", description = "Get the book by its id number")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.findById(id);
     }
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books",
+            description = "Get all books per website pages, default: page = 0, size = 20")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/search")
-    public List<BookDto> search(BookSearchParameters searchParameters) {
-        return bookService.search(searchParameters);
+    @Operation(summary = "Search books",
+            description = "Search books by title and author (titles=values&authors=values")
+    public List<BookDto> search(BookSearchParameters searchParameters, Pageable pageable) {
+        return bookService.search(searchParameters, pageable);
     }
 
     @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create a new book", description = "Create a new book")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update book by id", description = "Update the book by its id number")
     public BookDto updateById(@PathVariable Long id,

```

Comment:
you may also validate such things as @PathVariable, @RequestParam etc.

it will look like: `@PathVariable @Positive Long id`

however in order to do so you need to annotate your controller with @Validated annotation

this is just FYI :)

you decide - to apply it or not